diagnostic/engine.go

```diff
@@ -144,112 +236,6 @@ func (c *conflict) addSimilarConflict(conflict conflict) {
 	c.similarConflicts = append(c.similarConflicts, &conflict)
 }
 
-// ConflictList stores a list of conflicts.
-type ConflictList struct {
-	conflicts  []conflict
-	NoGrouping bool // if set to true, conflicts are not grouped by nil path. (Example use case: no-infer unit tests)
```

Comment:
This is removed and put as a parameter to `Diagnostics` instead, since it only affects the final diagnostic generation logic, but not the other parts (conflict collection logic).